src/background.ts

```diff
@@ -1,3 +1,24 @@
 import React from "react";
 
-console.log("welcome to uplink from console");
\ No newline at end of file
+console.log("welcome to uplink from console");
+let links: string[] = ["String"];
+//Initialize populated array of strings for URLs
+
+chrome.contextMenus.removeAll(function() {
+    chrome.contextMenus.create({
+        id: "rightclickoption",
+        title: "Save PDF",
+        contexts:["link"],
+       });
+});
+
+chrome.contextMenus.onClicked.addListener(function(info,tab){
+    let link:string = info.linkUrl as string;
+    links.push(link);
+    if(links[0] == "String")
+    {
+        links.shift();
+    }
+    //Push URL to link array and removes the 'String' placeholder
+    console.log(links)
+});
```

Comment:
Looks like it works but it's not integrated into the project. The user will be able to right click anything and save the link without logging in.

Reply:
I think in another PR that logic can come in, but essentially viewState needs to be able a valid state for file transfer before right clicking can be toggled